File: leetcode5.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n=s.size();
        if (n==0) return "";
        int res_length=1,res_left=0,start=0,pivot,left,right;
        while(start<n-1){
            pivot=start+1;
            while (pivot<n && s[pivot]==s[start]) pivot++;
            left=start-1;right=pivot;
            while (left>=0 && right<=n-1){
                if (s[left]==s[right]){
                    left--;right++;
                }
                else
                    break;
            }
            start=pivot;
            if (res_length<right-left-1){
                res_length=right-left-1;res_left=left+1;
            }
        }
        return s.substr(res_left,res_length);
    }
};

Runtime: 4 ms, faster than 99.88% of C++ online submissions for Longest Palindromic Substring.
Memory Usage: 8.9 MB, less than 99.30% of C++ online submissions for Longest Palindromic Substring

File: leetcode5.py
class Solution(object):    def longestPalindrome(self, s):        """        :type s: str        :rtype: str        """        n=len(s)        if n==0:            return ""        num,i=1,0        res=s[0]        while i<n-1:            pivot=i+1            while pivot<n and s[i]==s[pivot]:                pivot+=1            temp=s[i:pivot] #temp str            left,right=i-1,pivot            while left>=0 and right<=n-1:                if s[left]==s[right]:                    temp=s[left]+temp+s[right]                else:                    break                left-=1                right+=1            i=pivot            if len(temp)>num:                num,res=len(temp),temp        return resinput="caba"s=Solution()print(s.longestPalindrome(input))